core/controllers/skill_mastery.py

```diff
@@ -33,6 +34,25 @@ class SkillMasteryDataHandler(base.BaseHandler):
     """
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {
+            'comma_separated_exp_ids': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            }
+        },
+        'PUT': {
+            'mastery_change_per_skill': {
+                'schema': {
+                    'type': 'object_dict',
+                    'validation_method': (
+                        domain_objects_validator.validate_params_dict)
```

Comment:
Why is this schema used?

Reply:
`mastery_change_per_skill` is stored as a dictionary of multiple key value pairs, each one representing a different skill and the corresponding degree of mastery. I was unable to find a domain object for that schema so I resorted to using `validate_params_dict` as a general method of validation.

Reply:
@vojtechjelinek @aks681

Reply:
I don't think that makes sense, you should properly validate all the files of this dictionary. `validate_params_dict` is used in a case where we don't know the structure at all, but here we should be able to understand the structure

Reply:
I understand the structure but I'm very confused on how to validate it under the current schema definition. `dict` requires static keys to define a `name` and associated schema value, but `mastery_change_per_skill` could potentially be any `skill_id`.

@vojtechjelinek PTAL

Reply:
Okay, then you might need to write a validator that will take a dict where keys need to have skill ID format and values follow some given format.